src/main/java/org/hejnaluk/metrotimetable/service/ParseTimetableService.java

```diff
@@ -109,66 +110,152 @@ public class ParseTimetableService {
     public static final int TRIP_DIRECTION_ID = 5;
 
     public static final String DELIMITER = ",";
-    public static final String FIRST_DIRECTION = "0";
-    public static final String SECOND_DIRECTION = "1";
     public static final String ERROR_READING_FILE_ERROR_MESSAGE = "Error reading file: {}";
     public static final String NEW_LINE_AND_TAB = "\n\t";
 
+    private static final Map<String, List<Stop>> cacheListStopByRouteIdAndDirectionId = new ConcurrentHashMap<>();
+
+    private static final Map<String, TreeMap<LocalTime, List<CompleteStop>>> routeIdDirectionCache = new ConcurrentHashMap<>();
```

Comment:
TreeMap is not thread-safe

https://www.baeldung.com/java-concurrent-skip-list-map